Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Lib/DriverScore.hs

```diff
@@ -124,6 +128,36 @@ eventPayloadHandler _ DST.OnRideCompletion {..} = do
           pure . maybe 0 (const 1) $ (.nightShiftCharge) =<< mbFareParam
         Nothing -> pure 0
 
+updateDailyStats :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r) => Id DP.Person -> Id DMOC.MerchantOperatingCity -> DR.Ride -> m ()
+updateDailyStats driverId merchantOpCityId ride = do
+  transporterConfig <- CTCQ.findByMerchantOpCityId merchantOpCityId >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
+  localTime <- getLocalCurrentTime transporterConfig.timeDiffFromUtc
+  ds <- SQDS.findByDriverIdAndDate driverId (utctDay localTime)
+  case ds of
+    Nothing -> do
+      id <- generateGUIDText
+      let dailyStatsOfDriver' =
+            DDS.DailyStats
+              { id = id,
+                driverId = driverId,
+                totalEarnings = fromMaybe 0 ride.fare,
+                numRides = 1,
+                totalDistance = fromMaybe 0 ride.chargeableDistance,
+                merchantLocalDate = utctDay localTime
+              }
+      SQDS.create dailyStatsOfDriver'
+    Just dailyStats -> do
+      let dailyStatsOfDriver' =
+            DDS.DailyStats
+              { id = dailyStats.id,
+                driverId = driverId,
+                totalEarnings = dailyStats.totalEarnings + fromMaybe 0 ride.fare,
+                numRides = dailyStats.numRides + 1,
+                totalDistance = dailyStats.totalDistance + fromMaybe 0 ride.chargeableDistance,
+                merchantLocalDate = utctDay localTime
+              }
+      SQDS.upsertByDriverId dailyStatsOfDriver'
```

Comment:
name it update instead of upsert, its not doing upsert now. Also why send the whole driverStats object, just send what to update as args.

Reply:
done